CMakeLists.txt

```diff
@@ -159,8 +159,51 @@ ADD_CUSTOM_TARGET(setup_tests
 SET(ASPECT_TEST_GENERATOR "Unix Makefiles" CACHE STRING 
   "Generator to use for the test cmake project. Using ninja instead of make is not recommended.")
 
+
+# Allow us to make a global "test" target:
+IF(POLICY CMP0037)
+  # allow to override "test" target
+  CMAKE_POLICY(SET CMP0037 OLD)
+ENDIF()
+
+# This is an empty target but we will make it depend on tests/ and unit_tests/ next:
+ADD_CUSTOM_TARGET(test)
+
+# Generate CTestTestfile.cmake in the main build folder that lists all subfolders
+# that contain tests. This way you can call "ctest" in the build directory.
+FILE(WRITE ${CMAKE_BINARY_DIR}/CTestTestfile.cmake "# auto-generated ctest file\n")
+
+IF(EXISTS ${CMAKE_SOURCE_DIR}/unit_tests/CMakeLists.txt)
+  # If we have the unit_tests directory, add it so ctest picks it up and configure
+  # the test project in the subfolder:
+  FILE(APPEND ${CMAKE_BINARY_DIR}/CTestTestfile.cmake "SUBDIRS(unit_tests)\n")
+  FILE(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/unit_tests)
+  EXECUTE_PROCESS(
+    COMMAND ${CMAKE_COMMAND}
+	-D CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+	-D ASPECT_BINARY=${CMAKE_BINARY_DIR}/aspect
+	${CMAKE_CURRENT_SOURCE_DIR}/unit_tests
+    OUTPUT_FILE setup_unit_tests.log
+    RESULT_VARIABLE test_cmake_result
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/unit_tests
+   )
+  IF(NOT test_cmake_result EQUAL 0)
+    MESSAGE(FATAL_ERROR "ERROR: unittest/ project could not be configured.")
+  ENDIF()
+
+  # Finally hook up that "make test" will execute the tests:
+  ADD_CUSTOM_TARGET(run_unit_tests
+    COMMAND ${CMAKE_BINARY_DIR}/aspect --test
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/unit_tests
+    DEPENDS aspect
+    COMMENT "Running unit_tests ...")
+
+  ADD_DEPENDENCIES(test run_unit_tests)
```

Comment:
Ah, hum -- so calling `make test` checks the unit tests, but not the whole rest? I would have hoped that we can treat the unit tests the same way as all other tests -- i.e., by creating a particular test the same way we do for all of the .prm-based tests, just executing aspect differently.

Reply:
it is more complicated but I don't have a better way:
"make test" depends on "make run_unit_tests" and "make run_tests" (see the two calls to add_dependencies). They call "cmake build --target test" in the subdirectories.
Alternatively, you can call "ctest" in one of the subdirectories to run either the unit-tests or the other ones.
Finally, if you run ctest in the base directory (your build directory), the CTestTest.cmake file will let ctest to pick up tests in both subdirectories...